src/java.desktop/macosx/classes/com/apple/laf/AquaFileSystemModel.java

```diff
@@ -279,15 +279,15 @@ protected void sort(final Vector<Object> v) {
     // @param lo0 left boundary of array partition
     // @param hi0 right boundary of array partition
     abstract static class QuickSort {
-        final void quickSort(final Vector<Object> v, final int lo0, final int hi0) {
+        final void quickSort(final ArrayList<SortableFile> v, final int lo0, final int hi0) {
             int lo = lo0;
             int hi = hi0;
             SortableFile mid;
 
             if (hi0 > lo0) {
                 // Arbitrarily establishing partition element as the midpoint of
                 // the array.
-                mid = (SortableFile)v.elementAt((lo0 + hi0) / 2);
+                mid = v.get((lo0 + hi0) / 2);
```

Comment:
BTW integer overflow is possible here, which can result in `IndexOutOfBoundsException`